modules/transactions/src/integrationTest/java/org/apache/ignite/internal/tx/ItTransactionMetricsTest.java

```diff
@@ -0,0 +1,256 @@
+     */
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testCommitReadWriteTransaction(boolean implicit) {
+        Map<String, Long> metrics0 = metricValues(0);
+        Map<String, Long> metrics1 = metricValues(1);
+
+        Transaction tx = implicit ? null : node(0).transactions().begin();
+        keyValueView(0).put(tx, 12, "value");
+        if (!implicit) {
+            tx.commit();
+        }
+
+        Map<String, Long> actualMetrics0 = metricValues(0);
+        Map<String, Long> actualMetrics1 = metricValues(1);
+
+        // Check that there are no updates on the node 1.
+        testMetricValues(metrics1, actualMetrics1);
+
+        // Check that all transaction metrics ere not changed except TotalCommits and RwCommits.
+        testMetricValues(metrics0, actualMetrics0, "TotalCommits", "RwCommits");
+
+        assertThat(actualMetrics0.get("TotalCommits"), is(metrics0.get("TotalCommits") + 1));
+        assertThat(actualMetrics0.get("RwCommits"), is(metrics0.get("RwCommits") + 1));
+    }
+
+    /**
+     * Tests that TotalCommits and RwCommits/RoCommits are incremented when an "empty" transaction successfully committed.
+     * Empty means that there are no entries enlisted into the transaction.
+     */
+    @Test
+    void testCommitEmptyTransaction() {
+        Map<String, Long> metrics0 = metricValues(0);
+        Map<String, Long> metrics1 = metricValues(1);
+
+        Transaction rwTx = node(0).transactions().begin();
+        rwTx.commit();
+
+        Transaction roTx = node(0).transactions().begin(new TransactionOptions().readOnly(true));
+        roTx.commit();
+
+        Map<String, Long> actualMetrics0 = metricValues(0);
+        Map<String, Long> actualMetrics1 = metricValues(1);
+
+        // Check that there are no updates on the node 1.
+        testMetricValues(metrics1, actualMetrics1);
+
+        // Check that all transaction metrics ere not changed except TotalCommits, RwCommits and RoCommits.
+        testMetricValues(metrics0, actualMetrics0, "TotalCommits", "RwCommits", "RoCommits");
+
+        assertThat(actualMetrics0.get("TotalCommits"), is(metrics0.get("TotalCommits") + 2));
+        assertThat(actualMetrics0.get("RwCommits"), is(metrics0.get("RwCommits") + 1));
+        assertThat(actualMetrics0.get("RoCommits"), is(metrics0.get("RoCommits") + 1));
+    }
+
+    /**
+     * Tests that TotalRollbacks and RwRollbacks/RoRollbacks are incremented when a transaction rolled back.
+     */
+    @Test
+    void testRollbackTransaction() {
+        Map<String, Long> metrics0 = metricValues(0);
+        Map<String, Long> metrics1 = metricValues(1);
+
+        Transaction rwTx = node(0).transactions().begin();
+        keyValueView(0).put(rwTx, 12, "value");
+        rwTx.rollback();
+
+        Transaction roTx = node(0).transactions().begin(new TransactionOptions().readOnly(true));
+        keyValueView(0).get(roTx, 12);
+        roTx.rollback();
```

Comment:
I'd also check that total RW/RO are incremented in case of rollback caused by exception and timeout.